### ISSUE TITLE
Figure out how to make the client .re.mdx extension work.

### ISSUE THREAD

**Issue by cpojer** [nkzw-tech/remdx#2]
This should work using TypeScript reference types to `declare module "*.re.mdx"` but somehow it doesn't get picked up in the example project.

**Comment by karlhorky** [comment]
This sounds interesting for [my multi-slide deck implementation](https://github.com/nkzw-tech/remdx/issues/11) 🤔

But I'm not sure I'm understanding completely. What does adding types for the `.re.mdx` files enable? Are you somehow running TS type checking on your MDX files? Or is it something like the [IntelliSense enabled by `mdx-analyzer` / the MDX VS Code extension](https://github.com/microsoft/TypeScript/issues/36440#issuecomment-1898411970)?

It looks like this is the workaround for now, to have a separate `slides.re.mdx.d.ts` file. But if possible I'd like to avoid this from my multi-slide deck setup:

https://github.com/nkzw-tech/remdx/blob/851b854ff0db2095709dab7f157752cef44794b3/packages/create-remdx/template/slides.re.mdx.d.ts#L1-L11

**Comment by cpojer** [comment]
The point of these files is so that when you import a `.re.mdx` file it will be typed correctly by TypeScript so that it can be passed to the `render` function of ReMDX or transformed in other ways in a type safe way.

Normally it's possible to extend the extensions that TypeScript is aware of by providing these types from within the remdx package, similar to Vite's client types: https://vitejs.dev/guide/features#client-types

I tried various solutions but for some reason I could never figure out a way to just make it work and I ran out of time. As a "hack" I opted to put the `.d.ts` files into the template, so right now you'd need one of these for each slide deck file. It's not great, and I know it can be solved, but I just didn't get to it. Feel free to send a PR!
